### PR TITLE
curvefs add removexattr and enable acl

### PR DESCRIPTION
**Pull request by SeanHai**
### What problem does this PR solve?

Issue Number: #xxx 

Problem Summary:

### What is changed and how it works?

What's Changed:

How it Works:

Side effects(Breaking backward compatibility? Performance regression?):

### Check List

- [ ] Relevant documentation/comments is changed or added
- [ ] I acknowledge that all my contributions will be made under the project's license

### CODE REVIEW

**Comment by SeanHai** [comment]
cicheck

**Comment by SeanHai** [comment]
cicheck

**Comment by SeanHai** [comment]
cicheck

**Comment by SeanHai** [comment]
cicheck

**Comment by SeanHai** [comment]
cicheck

File: curvefs/src/client/fuse_client.cpp [hunk 1]
```diff
@@ -1149,8 +1149,38 @@ CURVEFS_ERROR FuseClient::FuseOpSetXattr(fuse_req_t req, fuse_ino_t ino,
     return CURVEFS_ERROR::OK;
 }
 
+CURVEFS_ERROR FuseClient::FuseOpRemoveXattr(fuse_req_t req, fuse_ino_t ino,
+    const char *name) {
+    std::string strname(name);
+    VLOG(1) << "FuseOpRemoveXattr ino: " << ino << ", name: " << name;
+
+    if (option_.fileSystemOption.disableXAttr || IsSpecialXAttr(name)) {
+        return CURVEFS_ERROR::NODATA;
```
Comment (h0hmj) [inline curvefs/src/client/fuse_client.cpp:1158]: is CURVEFS_ERROR::NODATA appropriate?

normally, user can only add/remove/modify "user.\*" xattr in my mind. in curvefs, maybe some "curve.*" in addtion. 

Some error code should be permission denied?

**Review by h0hmj — APPROVED** [review]

File: curvefs/conf/client.conf [hunk 2]
```diff
@@ -103,6 +103,8 @@ fuseClient.maxDataSize=1024
 # default refresh data interval 30s
 fuseClient.refreshDataIntervalSec=30
 fuseClient.warmupThreadsNum=10
+# enable acl
+fuseClient.enableACL=true
```
Comment (201341) [inline curvefs/conf/client.conf:107]: If disableXattr is true, we still can't use ACL

File: curvefs/src/client/fuse_client.cpp [hunk 3]
```diff
@@ -1149,8 +1149,38 @@ CURVEFS_ERROR FuseClient::FuseOpSetXattr(fuse_req_t req, fuse_ino_t ino,
     return CURVEFS_ERROR::OK;
 }
 
+CURVEFS_ERROR FuseClient::FuseOpRemoveXattr(fuse_req_t req, fuse_ino_t ino,
+    const char *name) {
+    std::string strname(name);
+    VLOG(1) << "FuseOpRemoveXattr ino: " << ino << ", name: " << name;
+
+    if (option_.fileSystemOption.disableXAttr || IsSpecialXAttr(name)) {
```
Comment (Wine93) [inline curvefs/src/client/fuse_client.cpp:1157]: We should handle the curve specified xattr even if the `disableXAttr` is true, so maybe the judge is wrong.

File: curvefs/src/client/fuse_client.cpp (hunk 1 above)
Comment (SeanHai) [reply curvefs/src/client/fuse_client.cpp:1158]: fixed

File: curvefs/conf/client.conf (hunk 2 above)
Comment (SeanHai) [reply curvefs/conf/client.conf:107]: The current logic is OK. ACL-related xattr is a special xattr and is not affected by the disablexattr switch.

File: curvefs/src/client/fuse_client.cpp (hunk 3 above)
Comment (SeanHai) [reply curvefs/src/client/fuse_client.cpp]: fixed

**Comment by SeanHai** [comment]
cicheck

**Comment by SeanHai** [comment]
cicheck

**Comment by SeanHai** [comment]
cicheck

**Comment by SeanHai** [comment]
cicheck

**Comment by SeanHai** [comment]
cicheck

**Review by Wine93 — APPROVED** [review]
LGTM!

**Comment by YunhuiChen** [comment]
ckcheck

**Pull request closed without merging**